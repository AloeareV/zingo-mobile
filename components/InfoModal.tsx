/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {View, ScrollView, SafeAreaView, Image} from 'react-native';
import {PrimaryButton, RegText} from './Components';
import {useTheme} from '@react-navigation/native';
import {Info} from 'app/AppState';

type DetailLineProps = {
  label: string;
  value?: string | number;
};
const DetailLine: React.FunctionComponent<DetailLineProps> = ({label, value}) => {
  const {colors} = useTheme();

  return (
    <View style={{display: 'flex', marginTop: 20}}>
      <RegText>{label}</RegText>
      <RegText color={colors.primary}>{value}</RegText>
    </View>
  );
};

type InfoModalProps = {
  info: Info | null;
  closeModal: () => void;
};

const InfoModal: React.FunctionComponent<InfoModalProps> = ({info, closeModal}) => {
  const {colors} = useTheme();
  const url = 'https://lightwalletd.zecwallet.co:1443';
  const height = info?.latestBlock;

  return (
    <SafeAreaView
      style={{
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'stretch',
        height: '100%',
        backgroundColor: colors.background,
      }}>
      <ScrollView
        style={{maxHeight: '85%'}}
        contentContainerStyle={{
          flexDirection: 'column',
          alignItems: 'stretch',
          justifyContent: 'flex-start',
          margin: 10,
        }}>
        <View style={{display: 'flex', alignItems: 'center'}}>
          <Image
            source={require('../assets/img/logobig.png')}
            style={{width: 100, height: 100, resizeMode: 'contain'}}
          />
        </View>

        <DetailLine label="Wallet Version" value="Zecwallet Lite v0.1" />
        <DetailLine label="Server Version" value={info?.version} />
        <DetailLine label="Lightwallet Server URL" value={url} />
        <DetailLine label="Network" value={info?.testnet ? 'Testnet' : 'Mainnet'} />
        <DetailLine label="Server Block Height" value={height} />
      </ScrollView>

      <View style={{flexGrow: 1, flexDirection: 'row', justifyContent: 'center', alignItems: 'center'}}>
        <PrimaryButton title="Close" onPress={closeModal} />
      </View>
    </SafeAreaView>
  );
};

export default InfoModal;
