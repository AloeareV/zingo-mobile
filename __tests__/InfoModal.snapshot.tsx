/**
 * @format
 */

import 'react-native';
import React from 'react';

import { create } from 'react-test-renderer';
import InfoModal from '../components/InfoModal';

jest.useFakeTimers()

// test suite
describe("Component InfoModal - test", () => {
  //snapshot test
  test("Matches the snapshot InfoModal", () => {
    const infoModal = create(<InfoModal info={null} closeModal={() => {}} />);
    expect(infoModal.toJSON()).toMatchSnapshot();
  });

});
