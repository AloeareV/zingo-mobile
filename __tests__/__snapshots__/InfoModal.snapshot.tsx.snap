// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InfoModal - test Matches the snapshot InfoModal 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    style={
      Object {
        "maxHeight": "85%",
      }
    }
  >
    <View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(255, 255, 255)",
              "paddingBottom": 25,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 28,
                "marginTop": 5,
                "padding": 5,
              }
            }
          >
            Server Info
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "marginTop": -25,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/logobig.png",
              }
            }
            style={
              Object {
                "height": 50,
                "resizeMode": "contain",
                "width": 50,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "margin": 10,
          }
        }
      >
        <View
          style={
            Object {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            Wallet Version
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 0.87,
                },
              ]
            }
          >
            zecwallet DAO v1.7.8
          </Text>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            Server Version
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 0.87,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            Lightwallet Server URL
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 0.87,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            Network
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 0.87,
                },
              ]
            }
          >
            Mainnet
          </Text>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            Server Block Height
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 0.87,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            ZEC Price
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 0.87,
                },
              ]
            }
          >
            -
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "rgb(28, 28, 30)",
          "borderRadius": 10,
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(28, 28, 30)",
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Close
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
